File: app/src/main/java/com/mismira/quickbeauty/BeautyAdjustParams.java
package com.mismira.quickbeauty;

/**
 * 酉고떚 蹂댁젙 ?뚮씪誘명꽣 紐⑤뜽 (0 ~ 100)
 */
public class BeautyAdjustParams {
    private int coolTone;   // 荑⑦넠 ?쇰? ?ㅼ뾽 (0 ~ 100)
    private int faceSlim;   // ?쇨뎬 ?ш린 異뺤냼 (0 ~ 100)
    private int chinSlim;   // ?깆꽑 V?쇱씤 蹂댁젙 (0 ~ 100)
    private int bodySlim;   // 紐몃ℓ ?꾩껜 ?щ┝ (0 ~ 100)

    public BeautyAdjustParams() {
        this.coolTone = 0;
        this.faceSlim = 0;
        this.chinSlim = 0;
        this.bodySlim = 0;
    }

    public BeautyAdjustParams(int coolTone, int faceSlim, int chinSlim, int bodySlim) {
        this.coolTone = clamp(coolTone);
        this.faceSlim = clamp(faceSlim);
        this.chinSlim = clamp(chinSlim);
        this.bodySlim = clamp(bodySlim);
    }

    public BeautyAdjustParams copy() {
        return new BeautyAdjustParams(coolTone, faceSlim, chinSlim, bodySlim);
    }

    public int getCoolTone() {
        return coolTone;
    }

    public void setCoolTone(int coolTone) {
        this.coolTone = clamp(coolTone);
    }

    public int getFaceSlim() {
        return faceSlim;
    }

    public void setFaceSlim(int faceSlim) {
        this.faceSlim = clamp(faceSlim);
    }

    public int getChinSlim() {
        return chinSlim;
    }

    public void setChinSlim(int chinSlim) {
        this.chinSlim = clamp(chinSlim);
    }

    public int getBodySlim() {
        return bodySlim;
    }

    public void setBodySlim(int bodySlim) {
        this.bodySlim = clamp(bodySlim);
    }

    public boolean isDefault() {
        return coolTone == 0 && faceSlim == 0 && chinSlim == 0 && bodySlim == 0;
    }

    public void reset() {
        this.coolTone = 0;
        this.faceSlim = 0;
        this.chinSlim = 0;
        this.bodySlim = 0;
    }

    private static int clamp(int val) {
        if (val < 0) return 0;
        if (val > 100) return 100;
        return val;
    }
}

File: app/src/main/java/com/mismira/quickbeauty/BeautyAdjustParams.kt
package com.quickbeauty.app

/**
 * 뷰티 보정 파라미터 모델 (0 ~ 100)
 */
data class BeautyAdjustParams(
    var coolTone: Int = 0,     // 1. 쿨톤 피부
    var faceSize: Int = 0,     // 2. 얼굴 전체 크기 축소 (소두)
    var chinSlim: Int = 0,     // 3. 턱선 V라인 슬림
    var faceLength: Int = 0,   // 4. 얼굴 세로 길이 축소 (얼굴 짧게 / 하관 리프팅)
    var shoulder: Int = 0,     // 5. 어깨 넓히기 (직각 어깨)
    var bodySlim: Int = 0      // 6. 몸매 슬림
) {
    // 이전 버전 및 호환성을 위한 faceSlim alias
    var faceSlim: Int
        get() = faceSize
        set(value) {
            faceSize = value
        }

    init {
        coolTone = coolTone.coerceIn(0, 100)
        faceSize = faceSize.coerceIn(0, 100)
        chinSlim = chinSlim.coerceIn(0, 100)
        faceLength = faceLength.coerceIn(0, 100)
        shoulder = shoulder.coerceIn(0, 100)
        bodySlim = bodySlim.coerceIn(0, 100)
    }

    fun isDefault(): Boolean = coolTone == 0 && faceSize == 0 && chinSlim == 0 && faceLength == 0 && shoulder == 0 && bodySlim == 0

    fun hasAnyEffect(): Boolean = !isDefault()

    fun reset() {
        coolTone = 0
        faceSize = 0
        chinSlim = 0
        faceLength = 0
        shoulder = 0
        bodySlim = 0
    }

    fun set(c: Int, fs: Int, cs: Int, fl: Int, sh: Int, b: Int) {
        coolTone = c.coerceIn(0, 100)
        faceSize = fs.coerceIn(0, 100)
        chinSlim = cs.coerceIn(0, 100)
        faceLength = fl.coerceIn(0, 100)
        shoulder = sh.coerceIn(0, 100)
        bodySlim = b.coerceIn(0, 100)
    }

    // 5개 인자 호환용 오버로드
    fun set(c: Int, fs: Int, cs: Int, fl: Int, b: Int) {
        set(c, fs, cs, fl, 0, b)
    }

    // 4개 인자 호환용 오버로드
    fun set(c: Int, fs: Int, fl: Int, b: Int) {
        set(c, fs, 0, fl, 0, b)
    }
}

